src/GafferSceneUI/Inspector.cpp

```diff
@@ -301,6 +310,303 @@ void Inspector::editScopeInputChanged( const Gaffer::Plug *plug )
 	}
 }
 
+PathPtr Inspector::historyPath()
+{
+	return new Inspector::HistoryPath( this, new Context( *Context::current() ) );
+}
+
+//////////////////////////////////////////////////////////////////////////
+// HistoryPath
+//////////////////////////////////////////////////////////////////////////
+
+Inspector::HistoryPath::HistoryPath(
+	const InspectorPtr inspector,
+	ConstContextPtr context,
+	const std::string &path,
+	PathFilterPtr filter
+) :
+	Path( path, filter ),
+	m_inspector( inspector ),
+	m_context( context ),
+	m_nodeNameConnections()
+{
+	assert( m_inspector );
+
+	Context::Scope currentScope( m_context.get() );
+
+	SceneAlgo::History::ConstPtr history = m_inspector->history();
+
+	assert( history );
+	assert( history->scene );
+
+	m_scriptNode = history->scene->source<ScenePlug>()->ancestor<ScriptNode>();
+
+	assert( m_scriptNode );
+
+	if( !isLeaf() )
+	{
+		// These signals update the whole path, so we can reduce the number of signals
+		// by only adding them to the root.
+		m_inspectorDirtiedConnection = m_inspector->dirtiedSignal().connect(
+			boost::bind( &HistoryPath::inspectorDirtied, this, ::_1 )
+		);
+		m_pathChangedConnection = pathChangedSignal().connect(
+			boost::bind( &HistoryPath::pathChanged, this, ::_1 )
+		);
+	}
+
+	emitPathChanged();
+}
+
+Inspector::HistoryPath::~HistoryPath()
+{
+
+}
+
+void Inspector::HistoryPath::propertyNames( std::vector<InternedString> &names, const Canceller *canceller) const
+{
+	Path::propertyNames( names );
+
+	if( isLeaf() )
+	{
+		names.push_back( g_valuePropertyName );
+		names.push_back( g_operationPropertyName);
+		names.push_back( g_acquiredEditPropertyName );
+		names.push_back( g_editWarningPropertyName );
+	}
+}
+
+ConstRunTimeTypedPtr Inspector::HistoryPath::property( const InternedString &name, const Canceller *canceller) const
+{
+
+	if( isLeaf() && name == g_namePropertyName )
+	{
+		// Remove the plug name from the end.
+		using Tokenizer = boost::tokenizer<boost::char_separator<char>>;
+
+		std::string nodeString;
+
+		ConstStringDataPtr nameData = runTimeCast<const StringData>( Path::property( g_namePropertyName, canceller ) );
+		std::string name = nameData->readable();
+
+		Tokenizer tokens( name, boost::char_separator<char>( "." ) );
+
+		for( Tokenizer::const_iterator it = tokens.begin(), eIt = tokens.end(); it != eIt; ++it )
+		{
+			Tokenizer::const_iterator next = it;
+			next++;
+			if( next == tokens.end() )
+			{
+				continue;
+			}
+			if( nodeString.size() )
+			{
+				nodeString += ".";
+			}
+			nodeString += *it;
+		}
```

Comment:
I think this can be expressed more easily with an `rfind()` and a `substr()`.

Reply:
Fixed in e6bd27f7be95b6d09d20f43b212b2bbdac71302b

Reply:
Actually, I think this can be expressed even more simply - `it->history->scene->node()->relativeName( m_scriptNode.get() )`. We get the name of the plug and then do string operations to find the name of the node, when we could simply get the name of the node?

Reply:
Or, simpler, since `nameChangedSignal()` is being tracked by `_HistoryWindow`, maybe HistoryPath shouldn't provide a node name property, but should instead provide the NodePtr itself, and a column managed by `_HistoryWindow` should get the name?

Reply:
This would also let us drop `m_scriptNode` from the HistoryPath class.

Reply:
It also removes the need for getting the node via `descendant()` related methods which is nice. Fixed in 4e6aa477c3ca0fd3e322c3e884f4632f3ee6a330